hal/network/lwip/socket_hal.cpp

```diff
@@ -94,3 +94,14 @@ int sock_fcntl(int s, int cmd, ...) {
   va_end(vl);
   return lwip_fcntl(s, cmd, val);
 }
+
+int sock_poll(struct pollfd* fds, nfds_t nfds, int timeout) {
+  int ret = lwip_poll(fds, nfds, timeout);
```

Comment:
Should be fine to simply `return lwip_poll(fds, nfds, timeout);`. Same in https://github.com/particle-iot/device-os/pull/1895/files#diff-a0762610abcc5f13d78b64aeb1f49facR105